runtime/src/iree/hal/string_util.c

```diff
@@ -164,6 +177,36 @@ IREE_API_EXPORT iree_status_t iree_hal_parse_element_type(
 IREE_API_EXPORT iree_status_t iree_hal_format_element_type(
     iree_hal_element_type_t element_type, iree_host_size_t buffer_capacity,
     char* buffer, iree_host_size_t* out_buffer_length) {
+  const char* specialname;
+  switch (element_type) {
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_8_E5M2:
+      specialname = "f8E5M2";
+      break;
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_8_E4M3:
+      specialname = "f8E4M3";
+      break;
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_8_E5M2_FNUZ:
+      specialname = "f8E5M2FNUZ";
+      break;
+    case IREE_HAL_ELEMENT_TYPE_FLOAT_8_E4M3_FNUZ:
+      specialname = "f8E4M3FNUZ";
+      break;
+    default:
+      specialname = NULL;
+  }
+  if (specialname) {
+    int n = strlen(specialname);
+    if (n + 1 > buffer_capacity) {
+      return iree_make_status(IREE_STATUS_FAILED_PRECONDITION,
+                              "insufficient buffer capacity");
+    }
+    if (out_buffer_length) {
+      *out_buffer_length = n;
+    }
+    strcpy(buffer, specialname);
```

Comment:
use memcpy if you've already done the strlen above - or just use snprintf like below to ensure it has matching behavior